pkg/storage/stores/shipper/compactor/compactor.go

```diff
@@ -201,7 +215,7 @@ func (c *Compactor) init(storageConfig storage.Config, schemaConfig loki_storage
 	c.indexStorageClient = shipper_storage.NewIndexStorageClient(objectClient, c.cfg.SharedStoreKeyPrefix)
 	c.metrics = newMetrics(r)
 
-	if c.cfg.RetentionEnabled {
+	if c.cfg.RetentionEnabled && c.deleteMode == deletion.WholeStreamDeletion {
```

Comment:
We need to be careful here since the code under this condition also sets things up specific to retention.

Reply:
I've moved that condition to later on in that method.

Is the deletion functionality possible without retention? In other words can be separate them. Or do you always need retention enabled to do deletion?